pkg/cfn/builder/nodegroup.go

```diff
@@ -267,15 +269,37 @@ func newLaunchTemplateData(n *NodeGroupResourceSet) *gfn.AWSEC2LaunchTemplate_La
 			HttpPutResponseHopLimit: gfn.NewInteger(2),
 		},
 	}
+
 	if !api.HasMixedInstances(n.spec) {
 		launchTemplateData.InstanceType = gfn.NewString(n.spec.InstanceType)
+
+		if strings.HasPrefix(n.spec.InstanceType, "t") {
+			isTInstance = true
+		}
 	} else {
 		launchTemplateData.InstanceType = gfn.NewString(n.spec.InstancesDistribution.InstanceTypes[0])
+
+		for _, instanceType := range n.spec.InstancesDistribution.InstanceTypes {
+			if strings.HasPrefix(instanceType, "t") {
+				isTInstance = true
+			}
+		}
 	}
 	if n.spec.EBSOptimized != nil {
 		launchTemplateData.EbsOptimized = gfn.NewBoolean(*n.spec.EBSOptimized)
 	}
 
+	// Add T3 Unlimited setting only if nodegroup has T-type instances
+	if n.spec.CPUCredits != nil {
+		if isTInstance {
+			launchTemplateData.CreditSpecification = &gfn.AWSEC2LaunchTemplate_CreditSpecification{
+				CpuCredits: gfn.NewString(strings.ToLower(*n.spec.CPUCredits)),
+			}
+		} else {
+			logger.Warning("cpuCredits option ignored, nodegroup %s has no T3 instance types", n.nodeGroupName)
```

Comment:
At this point this configuration should have been validated so I would avoid checking this again here. Let's check it only during the validation phase and trust that if cpuCredits is set it should be set in the template too